astore/client/commands/commands.go

```diff
@@ -64,10 +66,29 @@ func NewRoot(base *client.BaseFlags) *Root {
 	}
 
 	rc.store.Register(&kcobra.FlagSet{FlagSet: rc.PersistentFlags()}, "")
+	rc.Command.PersistentFlags().StringVarP(&rc.outputFile, "meta-file", "m", "",
+		fmt.Sprintf("Meta-data output file. Supported formats: %s", marshal.Formats()))
+
 	return rc
 }
 
 func (rc *Root) StoreClient() (*astore.Client, error) {
+	if rc.outputFile != "" {
+		// check output file type is supported
+		marshaller := marshal.ByExtension(rc.outputFile)
+		if marshaller == nil {
+			return nil, fmt.Errorf("Output file extension not supported `%s`.  Supported formats: %s",
+				rc.outputFile, marshal.Formats())
+		}
+
+		// check that the destination is writable
+		file, err := os.Create(rc.outputFile)
```

Comment:
nit: I would personally remove this check, there's no guarantee that the destination will still be writable by the time we get to it, I would just handle the error if it happens there.

(that said, I don't feel strongly)

Reply:
The error is impossible to handle later.  Here at least we have a chance.

Consider the upload command, what will happen is that the upload to astore happens first, and then the attempt to the write the meta-data file fails.  The data uploaded to astore is immutable -- there is no "handling" that error.   Also the upload is an expensive operation (takes a long time), better to fail early.

Reply:
sounds good.